src/include/optimizer/optimizer.h

```diff
@@ -52,11 +53,14 @@ class Optimizer : public AbstractOptimizer {
 
   Optimizer();
 
+
   std::shared_ptr<planner::AbstractPlan> BuildPelotonPlanTree(
       const std::unique_ptr<parser::SQLStatementList> &parse_tree) override;
 
   void Reset() override;
 
+  concurrency::Transaction *txn = nullptr;
```

Comment:
I don't think the optimizer should have a txn data member. It should be passed in to the functions.

@hzxa21 What are the implications of this?